test/e2e/allocator_test.go

```diff
@@ -178,6 +183,30 @@ func TestAllocatorCrossNamespace(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func copyDefaultAllocatorClientSecret(t *testing.T, toNamespace string) {
+	kubeCore := framework.KubeClient.CoreV1()
+	clientSecret, err := kubeCore.Secrets(allocatorClientSecretNamespace).Get(allocatorClientSecretName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Could not retrieve default allocator client secret %s/%s: %v", allocatorClientSecretNamespace, allocatorClientSecretName, err)
+	}
+	newSecret := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       clientSecret.TypeMeta.Kind,
+			APIVersion: clientSecret.TypeMeta.APIVersion,
+		},
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   clientSecret.ObjectMeta.Name,
+			Labels: clientSecret.ObjectMeta.Labels,
+		},
+		Data: clientSecret.Data,
+	}
```

Comment:
Can newSecret be removed and only store clientSecret `kubeCore.Secrets(toNamespace).Create(clientSecret)`

Reply:
Nope, the returned secret includes some metadata that you can't include when creating the secret (I got an error that a secret to create can't include a resourceVersion). So I just kept the relevant fields I think.

Reply:
Then you just needs to set `Name` and `Data` on the new secret. `TypeMeta` is inferred from API and `SecretTypeOpaque` is the default secret type

Reply:
Yeah, I simplified the copying logic. I just needed to set the ResourceVersion and the Namespace correctly.